dipy/workflows/segment.py

```diff
@@ -368,3 +372,144 @@ def run(
+            iterations unless another stopping criterion is met.
+            Used only for 'hmrf' method.
+        out_dir : string, optional
+            Output directory.
+        out_tissue : string, optional
+            Name of the tissue volume to be saved.
+        out_pve : string, optional
+            Name of the pve volume to be saved.
+
+        REFERENCES
+        ----------
+        .. footbibliography::
+
+        """
+        io_it = self.get_io_iterator()
+
+        if not method or method.lower() not in ["hmrf", "dam"]:
+            logging.error(
+                f"Unknown method '{method}' for tissue extraction. "
+                "Choose '--method hmrf' (for T1w) or '--method dam' (for DWI)"
+            )
+            sys.exit(1)
+
+        prefix = "t1" if method.lower() == "hmrf" else "dwi"
+        for i, name in enumerate(self.flat_outputs):
+            if name.endswith("tissue_classified.nii.gz"):
+                self.flat_outputs[i] = name.replace(
+                    "tissue_classified.nii.gz", f"{prefix}_tissue_classified.nii.gz"
+                )
+            if name.endswith("tissue_classified_pve.nii.gz"):
+                self.flat_outputs[i] = name.replace(
+                    "tissue_classified_pve.nii.gz",
+                    f"{prefix}_tissue_classified_pve.nii.gz",
+                )
+
+        self.update_flat_outputs(self.flat_outputs, io_it)
+
+        for fpath, tissue_out_path, opve in io_it:
+            logging.info(f"Extracting tissue from {fpath}")
+
+            data, affine, img = load_nifti(fpath, return_img=True)
+
+            if method.lower() == "hmrf":
+                classifier = TissueClassifierHMRF()
+                _, segmentation_final, PVE = classifier.classify(
+                    data, nclass, beta, tolerance=tolerance, max_iter=max_iter
+                )
+
+                save_nifti(tissue_out_path, segmentation_final, affine)
+                save_nifti(opve, PVE, affine)
+
+            elif method.lower() == "dam":
+                if bvals_file is None or not os.path.isfile(bvals_file):
+                    logging.error("'--bvals filename' is required for 'dam' method")
+                    sys.exit(1)
+
+                bvals, _ = read_bvals_bvecs(bvals_file, None)
+                wm_mask, gm_mask = dam_classifier(
+                    data,
+                    bvals,
+                    wm_threshold=wm_threshold,
+                    b0_threshold=b0_threshold,
+                    low_signal_threshold=low_signal_threshold,
+                )
+                result = np.zeros(wm_mask.shape)
+                result[wm_mask] = 1
+                result[gm_mask] = 2
+                save_nifti(tissue_out_path, result, affine)
+                save_nifti(
+                    opve, np.stack([wm_mask, gm_mask], axis=-1).astype(np.int32), affine
+                )
+
+            logging.info(f"Tissue saved as {tissue_out_path}")
```

Comment:
Since we always save PVE, should we also mention that PVE is saved at this path?

Reply:
thanks, updated!